src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -1555,6 +1574,13 @@ private object ReadEnumValue(bool insideJsonObjectValue, IEdmEnumTypeReference e
             // Try to read a null value
             if (ODataJsonReaderCoreUtils.TryReadNullValue(this.JsonReader, this.JsonLightInputContext, targetTypeReference, validateNullValue, propertyName, isDynamicProperty))
             {
+                if (isTopLevelPropertyValue && this.MessageReaderSettings.LibraryCompatibility == ODataLibraryCompatibility.Version6)

```

Comment:
>this.MessageReaderSettings.LibraryCompatibility == ODataLibraryCompatibility.Version6 [](start = 47, length = 85)

What if isTopLevelPropertyValue == true and this.MessageReaderSettings.LibraryCompatibility != 6?  I assume we don't expect to get here in that case, but if we did, wouldn't we want to raise an error? I'm wondering if the LibraryCompatibility == 6 is required here...